linodecli/plugins/obj/__init__.py

```diff
@@ -589,5 +613,216 @@ def _configure_plugin(client: CLI):
     )
 
     if cluster:
-        client.config.plugin_set_value("cluster", cluster)
+        client.config.plugin_set_value(CLUSTER_KEY, cluster)
+
+    perform_key_cleanup = _default_text_input(  # pylint: disable=protected-access
+        "Perform automatic cleanup of old linode-cli generated Object Storage keys?"
+        " The cleanup will occur before any Object Storage operation,"
+        " at most once every 24 hours. (Y/n)",
+        default="y",
+        validator=lambda x: (
+            "Please enter 'y' or 'n'"
+            if x.lower() not in ("y", "n", "yes", "no")
+            else None
+        ),
+    )
+    perform_key_cleanup = (
+        "yes" if perform_key_cleanup.lower() in ("y", "yes") else "no"
+    )
+    client.config.plugin_set_value(PERFORM_KEY_CLEANUP_KEY, perform_key_cleanup)
+    if perform_key_cleanup == "yes":
+        key_lifespan = _default_text_input(  # pylint: disable=protected-access
+            "Linode-cli Object Storage key lifespan (e.g. `30d` for 30 days)",
+            default="30d",
+            validator=lambda x: (
+                "Please enter a valid time duration"
+                if parse_time(x) is None
+                else None
+            ),
+        )
+        client.config.plugin_set_value(KEY_LIFESPAN_KEY, key_lifespan)
+
+        key_rotation_period = _default_text_input(  # pylint: disable=protected-access
+            "Linode-cli Object Storage key rotation period (e.g. `10d` for 10 days)",
+            default="10d",
+            validator=lambda x: (
+                "Please enter a valid time duration"
+                if parse_time(x) is None
+                else None
+            ),
+        )
+        client.config.plugin_set_value(
+            KEY_ROTATION_PERIOD_KEY, key_rotation_period
+        )
+
+        cleanup_batch_size = _default_text_input(  # pylint: disable=protected-access
+            "Number of old linode-cli Object Storage keys to clean up at once",
+            default="10",
+            validator=lambda x: (
+                "Please enter a valid integer" if not x.isdigit() else None
+            ),
+        )
+        client.config.plugin_set_value(
+            KEY_CLEANUP_BATCH_SIZE_KEY, str(int(cleanup_batch_size))
+        )
+
     client.config.write_config()
+
+
+def _cleanup_keys(client: CLI, force_key_cleanup: bool) -> None:
+    """
+    Cleans up stale linode-cli generated object storage keys.
+    """
+    try:
+        if not _should_perform_key_cleanup(client):
```

Comment:
```suggestion
        if not _should_perform_key_cleanup(client) and not force_key_cleanup:
```

As a user I would expect `--force-key-cleanup` to override the config.

Reply:
I refactored the logic a little bit, now force_key_cleanup is checked inside _should_perform_key_cleanup function.